### PR TITLE
Service Manager and Event Manager compatibility

### PR DESCRIPTION
**Pull request by basz**
Tested against a test application. It works... :-)

note : need to remove dev composer stuff

able to quickly review @bakura10 ?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -23,13 +23,25 @@
...
-        "zendframework/zend-mvc": "~2.2",
-        "zendframework/zend-servicemanager": "~2.2",
-        "zendframework/zend-stdlib": "~2.2",
-        "slm/queue": "0.4.*",
-        "aws/aws-sdk-php-zf2": "2.*"
+        "php": "^5.5 || ^7.0",
+        "zendframework/zend-eventmanager": "^2.6 || ^3.0",
+        "zendframework/zend-servicemanager": "~2.7 || ^3.0",
+        "zendframework/zend-stdlib": "~2.2 || ^3.0",
+        "slm/queue": "dev-eventmanager-deprecations as 0.6.0",
```
Comment (bakura10) [inline composer.json]: Make sure to tag slm/queue first and remove that :)

File: composer.json [hunk 2]
```diff
@@ -23,13 +23,25 @@
...
+        "aws/aws-sdk-php-zf2": "dev-zf-sm3-zf-ev3-compatibility as 3.0.0"
```
Comment (bakura10) [inline composer.json]: Did you make a PR to this?

File: src/SlmQueueSqs/Factory/SqsQueueFactory.php [hunk 3]
```diff
@@ -15,21 +16,33 @@
...
+        $options = new SqsQueueOptions(isset($config[$requestedName]) ? $config[$requestedName] : []);
 
 
         return new SqsQueue($sqsClient, $options, $requestedName, $jobPluginManager);
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function createService(ServiceLocatorInterface $serviceLocator, $name = '', $requestedName = '')
```
Comment (bakura10) [inline src/SlmQueueSqs/Factory/SqsQueueFactory.php]: This is to provide compatibility with SM 2.x right?

File: composer.json (hunk 2 above)
Comment (basz) [inline composer.json]: https://github.com/aws/aws-sdk-php-zf2/pull/36

File: src/SlmQueueSqs/Factory/SqsQueueFactory.php (hunk 3 above)
Comment (basz) [inline src/SlmQueueSqs/Factory/SqsQueueFactory.php]: yes

**Comment by bakura10** [comment]
Looks good to me !

**Comment by basz** [comment]
thank you @bakura10 . once the PR for the aws zf2 module is merged I will (ask you to) tag 0.6 on all modules.

**Comment by bakura10** [comment]
No problem ;). Ping me once it's done!

**Comment by basz** [comment]
@bakura the aws PR was merged. Since I have no clue if they will tag it (doesn't look like it) i simply referred too dev-master for now. If you can live with that I invite you to merge and tag 0.6.0 please. For the doctrine package I simply copied the release notes in the release notes fields.

**Comment by bakura10** [comment]
I've bumped for a tag. Let's wait a few days and see if they tag it, otherwise I'll tag this one with dev-master as dependency (lathough I don't really like that for a tag release).

**Comment by basz** [comment]
ok sure. no problem here

**Comment by basz** [comment]
@bakura10 done

**Comment by bakura10** [comment]
Tagged as 0.6.0!

Thanks for your hard work :).

**Comment by basz** [comment]
thank you

**Pull request merged**